presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcMetadataSessionPropertiesProvider.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.jdbc;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.session.PropertyMetadata;
+import com.google.common.collect.ImmutableList;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static com.facebook.presto.spi.session.PropertyMetadata.booleanProperty;
+
+public class JdbcMetadataSessionPropertiesProvider
+        implements JdbcSessionPropertiesProvider
+{
+    private final List<PropertyMetadata<?>> sessionProperties;
+    private static final String JDBC_METADATA_CACHE_ENABLED = "jdbc_metadata_cache_enabled";
```

Comment:
Since the cache is always created, maybe this session property is better named - `use_jdbc_metadata_cache` ?

Reply:
sounds good. thanks